scripts/cipublish

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -e
+
+if [[ -n "${NASA_HSI_DEBUG}" ]]; then
+    set -x
+fi
+
+function usage() {
+    echo -n \
+        "Usage: $(basename "$0")
+Publish container images to Elastic Container Registry (ECR).
+"
+}
+
+if [[ -n "${GIT_COMMIT}" ]]; then
+    GIT_COMMIT="${GIT_COMMIT:0:7}"
+else
+    GIT_COMMIT="$(git rev-parse --short HEAD)"
+fi
+
+function amazon_ecr_login() {
+    # Retrieves a temporary authorization token that can be used to access
+    # Amazon ECR, along with the registry URL.
+    read -r AUTHORIZATION_TOKEN ECR_REGISTRY \
+        <<<"$(aws ecr get-authorization-token \
+            --output "text" \
+            --query "authorizationData[0].[authorizationToken, proxyEndpoint]")"
+
+    # The authorization token is base64 encoded, and we need to strip the
+    # protocol from the registry URL.
+    AUTHORIZATION_TOKEN="$(echo "${AUTHORIZATION_TOKEN}" | base64 -d)"
+    ECR_REGISTRY="${ECR_REGISTRY##*://}"
+
+    # Authenticate to the ECR registry. The authorization token is presented in
+    # the format user:password.
+    echo "${AUTHORIZATION_TOKEN##*:}" |
+        docker login \
+            --username "${AUTHORIZATION_TOKEN%%:*}" \
+            --password-stdin "${ECR_REGISTRY}"
+}
```

Comment:
This is a new function I developed. It transitions us off the [deprecated](https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login.html) `aws ecr get-login` command, fixes the `WARNING! Using -*** the CLI is insecure. Use --password-stdin.` warning, and allows us to infer the registry URL rather than maintaining a dedicated secret.